tracing/src/macros.rs

```diff
@@ -2987,6 +2987,191 @@ macro_rules! valueset {
     };
 }
 
+#[doc(hidden)]
+#[macro_export]
+macro_rules! valueset {
+
+    // === base case ===
+    (@ $fields:expr, { $(,)* $(($field:expr, $val:expr)),* $(,)* } $(,)*) => {{
+        use $crate::__macro_support::Callsite as _;
+        // create a dummy callsite to get a bogus field.
+        static __CALLSITE: $crate::callsite::DefaultCallsite = $crate::callsite2! {
+            name: "__fake_tracing_callsite",
+            kind: $crate::metadata::Kind::SPAN,
+            target: module_path!(),
+            level: $crate::Level::TRACE,
+            fields: $crate::fieldset!("__fake_tracing_field" = "fake_tracing_value"),
+        };
```

Comment:
Please suggest a better way to create a fake Field. I hate this

Reply:
An alternative could be some construction based on ArrayVec, we only push the fields that are found.

Reply:
Why does this need to be different from the original implementation before #3398?

Reply:
The original implementation iterates over fields in the span metadata. It doesn't construct the field name based on the actual current field given. `recordall!` simply never worked beyond the case of the exact same spans in the exact order.